src/connectionController.ts

```diff
@@ -116,142 +122,188 @@ export default class ConnectionController {
     this._telemetryService = telemetryService;
   }
 
-  async _migratePreviouslySavedConnection(
-    savedConnectionInfo: StoreConnectionInfo
-  ): Promise<StoreConnectionInfoWithConnectionOptions> {
-    if (!savedConnectionInfo.connectionModel) {
-      throw new Error(
-        'The connectionModel object is missing in saved connection info.'
+  async loadSavedConnections(): Promise<void> {
+    const globalAndWorkspaceConnections = {
+      ...this._storageController.get(
+        StorageVariables.GLOBAL_SAVED_CONNECTIONS,
+        StorageLocation.GLOBAL
+      ),
+      ...this._storageController.get(
+        StorageVariables.WORKSPACE_SAVED_CONNECTIONS,
+        StorageLocation.WORKSPACE
+      ),
+    };
+
+    let connectionsDidChange = false;
```

Comment:
We might also just check `Object.keys(this._connections).length`.

Reply:
Yea this should be good as well. I was probably trying to avoid a situation when `this._connections` has entries already but just checked again and it is very unlikely because it is called only once, when the extension activates.

Will be updating with the suggestion.